File: amazia-core/src/main/java/com/example/sku/controller/CreateProductSkuPriceController.java
package com.example.sku.controller;

import com.example.sku.entity.ProductSkuPrice;
import com.example.sku.service.CreateProductSkuPriceService;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

@RestController
@RequestMapping("/api")
public class CreateProductSkuPriceController {

    private final CreateProductSkuPriceService createProductSkuPriceService;

    public CreateProductSkuPriceController(CreateProductSkuPriceService createProductSkuPriceService) {
        this.createProductSkuPriceService = createProductSkuPriceService;
    }

    @PostMapping("/skus/{id}/prices")
    public ResponseEntity<ProductSkuPrice> create(@PathVariable Long id, @RequestBody ProductSkuPrice request) {
        return ResponseEntity.status(HttpStatus.CREATED).body(createProductSkuPriceService.create(id, request));
    }
}

File: amazia-console/app/Sku/Controller/CreateProductSkuPriceController.php
<?php

namespace App\Sku\Controller;

use App\Sku\Service\CreateProductSkuPriceService;
use Illuminate\Http\JsonResponse;
use Illuminate\Http\Request;
use Illuminate\Routing\Controller;

class CreateProductSkuPriceController extends Controller
{
    public function __construct(private CreateProductSkuPriceService $createProductSkuPriceService) {}

    public function __invoke(Request $request, int $id): JsonResponse
    {
        $response = $this->createProductSkuPriceService->create($id, $request->all());
        return response()->json($response->json(), $response->status());
    }
}
